sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -227,36 +218,30 @@ def _build_message(self, body):
     def _set_message_annotations(self, key, value):
         if not self._raw_amqp_message.annotations:
             self._raw_amqp_message.annotations = {}
-
-        if isinstance(self, ServiceBusReceivedMessage):
-            try:
-                del self._raw_amqp_message.annotations[key]
-            except KeyError:
-                pass
-
         if value is None:
             try:
-                del self._raw_amqp_message.annotations[ANNOTATION_SYMBOL_KEY_MAP[key]]
+                del self._raw_amqp_message.annotations[key]
             except KeyError:
                 pass
         else:
-            self._raw_amqp_message.annotations[ANNOTATION_SYMBOL_KEY_MAP[key]] = value
+            self._raw_amqp_message.annotations[key] = value
 
-    def _to_outgoing_message(self):
-        # type: () -> ServiceBusMessage
-        # pylint: disable=protected-access
-        self.message = self.raw_amqp_message._to_outgoing_amqp_message()
+    def _to_outgoing_message(self) -> "ServiceBusMessage":
         return self
 
     @property
-    def raw_amqp_message(self):
-        # type: () -> AmqpAnnotatedMessage
+    def message(self) -> LegacyMessage:
+        if not self._uamqp_message:
+            self._uamqp_message = LegacyMessage(self._raw_amqp_message)
```

Comment:
Since I've moved the Message creation in AmqpAnnotatedMessage (specifically, the `to_outgoing_amqp_message` method) out to a Amqp Transport object, I think we'll need to call that method like:
`to_outgoing_amqp_message(self._message)` rather than: `self._message._to_outgoing_amqp_message()`. So I was thinking we could just do the following here:
```suggestion
            self._uamqp_message = LegacyMessage(self._raw_amqp_message, to_outgoing_amqp_message=PyamqpTransport().to_outgoing_amqp_message)
```
Thoughts/concerns?